src/OpenTelemetry.Instrumentation.Wcf/Implementation/RequestTelemetryStateTracker.cs

```diff
@@ -139,8 +139,13 @@ public EntryTimeoutProperties(string messageId, long expiresAt, Action<RequestTe
             this.TimeoutCallback = timeoutCallback;
         }
 
-        public int CompareTo(object obj)
+        public int CompareTo(object? obj)
         {
+            if (obj == null)
```

Comment:
In your version (in theory) the cast could still blow up. I think this is slightly more correct:

```diff
-        public int CompareTo(object? obj)
+        public int CompareTo(object? obj)
         {
-            var other = (EntryTimeoutProperties)obj;
+            if (obj is not EntryTimeoutProperties other)
+            {
+                return -1; // Note sure if -1 is totally correct. 1 could be fine too
+            }
```

Reply:
Fixed by 80258e3baf9f7cc2e6627328e512a065ae04fb27

Based on what I have found here https://learn.microsoft.com/en-us/dotnet/api/system.icomparable.compareto?view=net-8.0&redirectedfrom=MSDN#examples returning 1 is the recommended option for `null` objects